downloader_test.go

```diff
@@ -415,6 +415,48 @@ func TestTransientDownstreamError(t *testing.T) {
 		}
 	}
 }
+
+func TestNoRetries(t *testing.T) {
```

Comment:
You should add a test that verifies 2 jobs with different max_retires are indeed requeued the respective amount

Reply:
This already tested in `TestPerformDownloadRequeue`, on processor tests. But there is no integration test for it, I'll add that.